src/cc/lib/graph/partition.cc

```diff
@@ -476,24 +476,44 @@ size_t Partition::FullNeighbor(uint64_t internal_id, std::span<const Type> edge_
         {
             const auto start = m_edge_type_offset[i];
             const auto last = m_edge_type_offset[i + 1];
-            result += last - start;
-
-            // m_edge_destination[last-1]+1 - take the last element and then advance the pointer
-            // to imitate std::end, otherwise we'll have an out of range exception.
-            out_neighbors_ids.insert(std::end(out_neighbors_ids), &m_edge_destination[start],
-                                     &m_edge_destination[last - 1] + 1);
-            auto original_type_size = out_edge_types.size();
-            out_edge_types.resize(original_type_size + last - start, m_edge_types[i]);
-            out_edge_weights.reserve(out_edge_weights.size() + last - start);
-            for (size_t index = start; index < last; ++index)
-            {
-                out_edge_weights.emplace_back(index > start ? m_edge_weights[index] - m_edge_weights[index - 1]
-                                                            : m_edge_weights[start]);
-            }
+
+            result += func(start, last, i);
         }
     }
     return result;
 }
+
+size_t Partition::NeighborCount(uint64_t internal_id, std::span<const Type> edge_types) const
+{
+    auto lambda = [&](auto start, auto last, int i) { return last - start; };
```

Comment:
please, don't capture everything by reference in lambda, but explicitly list items you'll use.

Reply:
Same below